packages/sdk-v2/src/sdk/savings/react.ts

```diff
@@ -0,0 +1,216 @@
+import React, { useCallback } from 'react'
+import { useContractFunction, useCalls, QueryParams, useEthers } from '@usedapp/core'
+import { useGetContract } from '../base/react'
+import { ethers } from 'ethers'
+import { EnvKey } from '../base'
+import { GoodDollarStaking, IGoodDollar } from '@gooddollar/goodprotocol/types'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { G$ } from '@gooddollar/web3sdk'
+import { getGasPrice } from '../constants'
+
+export interface StakerInfo {
+  claimable: {
+    g$Reward: CurrencyAmount<Currency> | any,
+    goodReward: CurrencyAmount<Currency> | any
+  } | undefined, 
+  balance?: CurrencyAmount<Currency> | any,
+  rewardsPaid?: {
+    g$Minted: CurrencyAmount<Currency> | any,
+    goodMinted: CurrencyAmount<Currency> | any,
+  }
+  deposit?: CurrencyAmount<Currency> | any,
+  principle?: CurrencyAmount<Currency> | any
+  shares?: number | any,
+  lastSharePrice?: number | any
+}
+
+export interface GlobalStats {
+  totalStaked: CurrencyAmount<Currency> | undefined,
+  totalRewardsPaid: CurrencyAmount<Currency> | undefined,
+  apy: number | undefined
+  lastUpdateBlock?: number,
+  savings?: number,
+}
+
+export function useSavingsBalance(refresh: QueryParams["refresh"] = "never", env?: EnvKey) {
+  const { account } = useEthers()
+  const gooddollar = useGetContract("GoodDollar", true, "savings", env) as IGoodDollar; 
+  const gdStaking = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking;
+
+  const results = useCalls(
+    [
+      {
+        contract: gooddollar,
+        method: 'balanceOf',
+        args: [account]
+      },
+      {
+        contract: gdStaking,
+        method: 'getSavings',
+        args: [account]
+      }
+    ]
+  )
+
+  if (results[0]?.error || results[1]?.error){
+  }
+  let g$Balance: string = '0', savingsBalance: string = '0';
+  if (!results.includes(undefined)) {
+    const [balance] = results[0]?.value
+    const [sBalance] = results[1]?.value
+    
+    g$Balance = parseFloat( (parseInt(balance) / 1e2).toString() ).toFixed(2)
+    savingsBalance = parseFloat( (parseInt(sBalance) / 1e2).toString()).toFixed(2)
```

Comment:
```> const savingsBalance = await contract.getSavings(signer.address)
undefined
> savingsBalance / 1e2
50020.84
> savingsBalance / 1e18
5.002084e-12
> ```

Reply:
I think that the logic for handling the amount of decimals etc should be in the SDK
so that anyone using the SDK can rely on what data is provided
I don't think that should be of any concern for the UI

Reply:
in other words, the view shouldnt be responsible for controller function imo

Reply:
number of decimals to display is a VIEW decision, it has nothing to do with control.
in anycase you can query the savings contract for .decimals() becuase it is also a token.

Reply:
okay, but if I provide those decimals with e18, the savingsbalance is not shown as should

Reply:
so you want nothing in the sdk about formatting, just return BigNumbers/CurrencyAmount and add this logic to UI?

Reply:
moved formatting of balance to UI